### PR TITLE
AssemblyResolve

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/Plugins/Plugin.cs [hunk 1]
```diff
@@ -85,16 +85,20 @@ private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEven
...
-            if (assembly != null)
-                return assembly;
+            if (assembly is null)
+                assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.GetName().Name == an.Name);
+            if (assembly != null) return assembly;
 
-            AssemblyName an = new AssemblyName(args.Name);
-            string filename = an.Name + ".dll";
+            string path = an.Name + ".dll";
+            if (!File.Exists(path)) path = Combine(PluginsDirectory, path);
```
Comment (shargon) [inline src/neo/Plugins/Plugin.cs]: But this path is the plugins path, we need the specific plugin
`Neo\Plugins\Rpc Wallet\Mi Dependency.dll`

Comment (erikzhang) [reply src/neo/Plugins/Plugin.cs]: You mean we should put the dependencies into the config directory? Then there is no way for plugins to share dependency files.

Comment (shargon) [reply src/neo/Plugins/Plugin.cs]: Then we need to release the plugin zip with this structure.

```
MyDependency.dll
MyPlugin\MyPlugin.dll
```

If we will release the plugins like this, is good to me.

File: src/neo/Plugins/Plugin.cs [hunk 2]
```diff
@@ -85,16 +85,20 @@ private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEven
             if (args.Name.Contains(".resources"))
                 return null;
 
+            AssemblyName an = new AssemblyName(args.Name);
+
             Assembly assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName == args.Name);
-            if (assembly != null)
-                return assembly;
+            if (assembly is null)
+                assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.GetName().Name == an.Name);
```
Comment (shargon) [inline src/neo/Plugins/Plugin.cs]: Can we join this two linq queries in only one?

Comment (erikzhang) [reply src/neo/Plugins/Plugin.cs]: No. They are two different queries.

**Comment by shargon** [comment]
Wait for https://github.com/neo-project/neo-modules/pull/160

**Comment by erikzhang** [comment]
Is it tested?

**Comment by cloud8little** [comment]
> Is it tested?

Yes, good to merge.

**Pull request merged**